### PR TITLE
Inline Rename: Map caret position to correct buffer on Home/End

### PR DESCRIPTION
**Pull request by dpoeschl**
Fixes bug #2516 

Tagging for Code Review: @Pilchie @jasonmalinowski @rchande @balajikris @brettfo @jmarolf

### CODE REVIEW

File: src/EditorFeatures/Core/Implementation/InlineRename/CommandHandlers/RenameCommandHandler_LineStartEndHandler.cs [hunk 1]
```diff
@@ -92,16 +92,22 @@ private bool HandleLineStartOrLineEndCommand(ITextBuffer subjectBuffer, ITextVie
                         return false;
                     }
 
+                    var newPointInView = view.BufferGraph.MapUpToBuffer(newPoint, PointTrackingMode.Negative, PositionAffinity.Successor, view.TextBuffer);
```
Comment (dpoeschl) [inline src/EditorFeatures/Core/Implementation/InlineRename/CommandHandlers/RenameCommandHandler_LineStartEndHandler.cs]: "Negative" and "Successor" chosen randomly, but they work...

**Comment by jmarolf** [comment]
:+1:

File: src/EditorFeatures/Core/Implementation/InlineRename/CommandHandlers/RenameCommandHandler_LineStartEndHandler.cs (hunk 1 above)
Comment (jasonmalinowski) [inline src/EditorFeatures/Core/Implementation/InlineRename/CommandHandlers/RenameCommandHandler_LineStartEndHandler.cs]: Please add a comment explaining your choice.

I would imagine that point tracking mode doesn't matter if you're not tracking between different versions. You'll probably want to think carefully about PositionAffinityMode though.

**Comment by jasonmalinowski** [comment]
:+1: once there's a comment explaining the choice of flags to the buffer graph mapping.

File: src/EditorFeatures/Core/Implementation/InlineRename/CommandHandlers/RenameCommandHandler_LineStartEndHandler.cs (hunk 1 above)
Comment (Pilchie) [inline src/EditorFeatures/Core/Implementation/InlineRename/CommandHandlers/RenameCommandHandler_LineStartEndHandler.cs]: You probably want `lineStart ? Successor : Predecessor`

**Comment by Pilchie** [comment]
:+1:

**Pull request merged**